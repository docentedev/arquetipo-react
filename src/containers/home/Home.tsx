import React, { useState, useEffect, Dispatch, SetStateAction } from 'react';
import { useStateManager } from 'use-state-manager';

import './Home.css';

import { yearManager } from '../../store/managers';

import { findAllUsersData } from './Home.utils';
import User from '../../types/user';
import Card from '../../components/card/Card';
import CardRow from '../../components/card-row/CardRow';
import DateDisplay from '../../components/date-display/DateDisplay';


const renderUser = (user : User) => (<Card key={user.id} user={user} />);

/**
 * @author Claudio Rojas <claudio.dcv@gmail.com>
 * @since v0.1.0
 * @version v1.0.0
 * @param props
 */
const Home: React.FC = () => {

  const [users, setUsers] : [Array<User>, Dispatch<SetStateAction<any>>] = useState([]);
  
  const [year, setYear] = useStateManager(yearManager);

  useEffect(() => {
    findAllUsersData().then(setUsers);
  }, []);

  return (
    <div className="home">
      {year}
      <button onClick={() => setYear(12)}>Cambiar a√±o</button>
      <DateDisplay />
      <CardRow>{users.map(renderUser)}</CardRow>
    </div>
  );
};
  
export default Home;