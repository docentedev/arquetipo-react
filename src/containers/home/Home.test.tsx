import React from 'react';
import Home from './Home';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

test('Home is ok renderer', () => {
    const component = renderer.create(
        <Home />,
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      const button = shallow((<Home />));
      button.find('button').simulate('click');

});