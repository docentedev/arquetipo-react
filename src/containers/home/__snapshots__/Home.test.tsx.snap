// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home is ok renderer 1`] = `
<div
  className="home"
>
  1990
  <button
    onClick={[Function]}
  >
    Cambiar a√±o
  </button>
  <div
    className="date-display"
  >
    12/19/2019
  </div>
  <div
    className="card-row"
  />
</div>
`;
